File: db4oj.tests/src/com/db4o/db4ounit/common/soda/arrays/STArrStringTTestCase.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.db4ounit.common.soda.arrays;

import com.db4o.db4ounit.common.soda.util.*;
import com.db4o.query.*;

public class STArrStringTTestCase extends SodaBaseTestCase {
	
	public static class Data {
		public String[] _strArr;

		public Data(String[] arr) {
			_strArr = arr;
		}
	}
	
	protected Object[] createData() {
		return new Object[]{
			new Data(null),
			new Data(new String[] {null}),
			new Data(new String[] {null, null}),
			new Data(new String[] {"foo", "bar", "fly"}),
			new Data(new String[] {null, "bar", "wohay", "johy"})
		};
	}
	
	public void testDefaultContainsOne(){
		Query q = newQuery();
		q.constrain(new Data(new String[] {"bar"}));
		expect(q, new int[] {3,4});
	}
	
	public void testDefaultContainsTwo(){
		Query q = newQuery();
		q.constrain(new Data(new String[] {"foo", "bar"}));
		expect(q, new int[] {3});
	}
	
	public void testDescendOne(){
		Query q = newQuery(Data.class);
		q.descend("_strArr").constrain("bar");
		expect(q, new int[] {3,4});
	}
	
	public void testDescendTwo(){
		Query q = newQuery(Data.class);
		Query qElements = q.descend("_strArr");
		qElements.constrain("foo");
		qElements.constrain("bar");
		expect(q, new int[] {3});
	}
	
	public void testDescendOneNot(){
		Query q = newQuery(Data.class);
		q.descend("_strArr").constrain("bar").not();
		expect(q, new int[] {0,1,2});
	}
	
	public void testDescendTwoNot(){
		Query q = newQuery(Data.class);
		Query qElements = q.descend("_strArr");
		qElements.constrain("foo").not();
		qElements.constrain("bar").not();
		expect(q, new int[] {0,1,2});
	}
}
	

File: db4on/test/src/com/db4o/db4ounit/common/soda/arrays/typed/STArrStringTTestCase.cs
namespace com.db4o.db4ounit.common.soda.arrays.typed
{
	public class STArrStringTTestCase : com.db4o.db4ounit.common.soda.util.SodaBaseTestCase
	{
		public string[] strArr;

		public STArrStringTTestCase()
		{
		}

		public STArrStringTTestCase(string[] arr)
		{
			strArr = arr;
		}

		public override object[] CreateData()
		{
			return new object[] { new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase
				(), new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase(new string
				[] { null }), new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase
				(new string[] { null, null }), new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase
				(new string[] { "foo", "bar", "fly" }), new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase
				(new string[] { null, "bar", "wohay", "johy" }) };
		}

		public virtual void TestDefaultContainsOne()
		{
			com.db4o.query.Query q = NewQuery();
			q.Constrain(new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase(new 
				string[] { "bar" }));
			Expect(q, new int[] { 3, 4 });
		}

		public virtual void TestDefaultContainsTwo()
		{
			com.db4o.query.Query q = NewQuery();
			q.Constrain(new com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase(new 
				string[] { "foo", "bar" }));
			Expect(q, new int[] { 3 });
		}

		public virtual void TestDescendOne()
		{
			com.db4o.query.Query q = NewQuery();
			q.Constrain(typeof(com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase)
				);
			q.Descend("strArr").Constrain("bar");
			Expect(q, new int[] { 3, 4 });
		}

		public virtual void TestDescendTwo()
		{
			com.db4o.query.Query q = NewQuery();
			q.Constrain(typeof(com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase)
				);
			com.db4o.query.Query qElements = q.Descend("strArr");
			qElements.Constrain("foo");
			qElements.Constrain("bar");
			Expect(q, new int[] { 3 });
		}

		public virtual void TestDescendOneNot()
		{
			com.db4o.query.Query q = NewQuery();
			q.Constrain(typeof(com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase)
				);
			q.Descend("strArr").Constrain("bar").Not();
			Expect(q, new int[] { 0, 1, 2 });
		}

		public virtual void TestDescendTwoNot()
		{
			com.db4o.query.Query q = NewQuery();
			q.Constrain(typeof(com.db4o.db4ounit.common.soda.arrays.typed.STArrStringTTestCase)
				);
			com.db4o.query.Query qElements = q.Descend("strArr");
			qElements.Constrain("foo").Not();
			qElements.Constrain("bar").Not();
			Expect(q, new int[] { 0, 1, 2 });
		}
	}
}
